src/Quest.sol

```diff
@@ -74,24 +74,35 @@ contract Quest is ERC721 {
     ////////// Constructor //////////
     /////////////////////////////////
 
+    constructor(
+        string memory _name,
+        string memory _symbol
+    ) ERC721(_name, _symbol) {}
+
+    // @notice Initialize the contract after it has been deployed
     /// @param _name Name of the Quest NFT
     /// @param _symbol Symbol for the Quest NFT
     /// @param _contributors List of addresses to pre-mint to
     /// @param _tokenURI URI for the token
     /// @param _contractURI Contract metadata URI (for NFT marketplaces)
-    constructor(
+    function initialize(
```

Comment:
🚨

There's a big security issue here. Currently, this `initialize` function is not guarded in any way, so anyone can call it after quest creation, allowing them to change token metadata and distribute to additional contributors.

The [OZ Initializable contract](https://docs.openzeppelin.com/contracts/4.x/api/proxy#Initializable) should be inherited and the `initializer` modifier added to the `initialize` function to guard against repeated initializations.

Reply:
Does it also make sense to limit who can call `initialize` to the factory somehow?

Reply:
👍